src/functional/ffront/past_to_itir.py

```diff
@@ -56,8 +62,8 @@ class ProgramLowering(NodeTranslator):
     contexts = (SymbolTableTrait.symtable_merger,)
 
     @classmethod
-    def apply(cls, node: past.Program) -> itir.FencilDefinition:
-        return cls().visit(node)
+    def apply(cls, node: past.Program, **kwargs) -> itir.FencilDefinition:
```

Comment:
Should `function_definitions` be an explicit parameter here?

Reply:
Done.